perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java

```diff
@@ -539,20 +540,66 @@ public void checkReservedUnixGroupNames(Attribute groupNameAttribute) throws Wro
 	}
 
 	@Override
-	public void checkUnpermittedUserLogins(Attribute loginAttribute) throws WrongAttributeValueException {
-		if(loginAttribute == null) return;
+	public void checkLoginNamespaceRegex(String namespace, String login, Pattern defaultRegex) throws InvalidLoginException {
+		Utils.notNull(namespace, "namespace to check login syntax");
+		Utils.notNull(login, "login to check syntax for");
+
 		checkPerunNamespacesMap();
 
-		String unpermittedNames = perunNamespaces.get(loginAttribute.getFriendlyName() + ":reservedNames");
-		if (unpermittedNames != null) {
-			List<String> unpermittedNamesList = Arrays.asList(unpermittedNames.split("\\s*,\\s*"));
-			if (unpermittedNamesList.contains(loginAttribute.valueAsString()))
-				throw new WrongAttributeValueException(loginAttribute, "This login is not permitted.");
+		String regex = perunNamespaces.get("login-namespace:"+namespace+":regex");
+		if (regex != null) {
+			//Check if regex is valid
+			try {
+				Pattern.compile(regex);
+			} catch (PatternSyntaxException e) {
+				log.error("Regex pattern \"{}\" from \"login-namespace:{}:regex\" property of perun-namespaces.properties file is invalid.", regex, namespace);
+				throw new InternalErrorException("Regex pattern \""+regex+"\" from \"login-namespace:"+namespace+":regex\" property of perun-namespaces.properties file is invalid.");
+			}
+			// check syntax or if its between exceptions
+			if(!login.matches(regex) && !isLoginException(namespace, login)) {
+				log.warn("Login '{}' in {} namespace doesn't match regex: {}", login, namespace, regex);
+				throw new InvalidLoginException("Login doesn't matches expected regex: \"" + regex +"\"");
+			}
+		} else {
+			// Regex property not found in our attribute map, so use the default hardcoded regex
+			// check syntax or if its between exceptions
+			if (!defaultRegex.matcher(login).matches() && !isLoginException(namespace, login)) {
+				log.warn("Login '{}' in {} namespace doesn't match regex: {}", login, namespace, regex);
+				throw new InvalidLoginException("Login doesn't matches expected regex: \"" + defaultRegex +"\"");
+			}
+		}
+	}
+
+	@Override
+	public boolean checkIfUserLoginIsPermitted(String namespace, String login) {
```

Comment:
I was thinking about this method and also there would be better name something like "isUserLoginPermitted", because it returns true or false. Check is more like OK or Exception in our code.

Reply:
Renamed, makes sense.